discord/src/DiscordBot.jl

```diff
@@ -0,0 +1,84 @@
+module DiscordBot
+
+using Dates: DateTime, Millisecond, UTC, now, @dateformat_str
+
+using Discord: BotClient, create_message
+using JSON3: JSON3
+
+const DATA_DIR = joinpath(@__DIR__, "..", "..", "data")
+const DATETIME_FORMAT = dateformat"yyyy-mm-ddTHH:MM:SSZ"
+
+discord_client() = BotClient(ENV["DISCORD_TOKEN"])
+
+function load_channels()
+    data = open(JSON3.read, joinpath(DATA_DIR, "discord.json"))
+    return Dict(ch.name => (; id=ch.channelId, color=ch.color) for ch in data.channels)
+end
+
+function load_talks()
+    talks = open(JSON3.read, joinpath(DATA_DIR, "talks.json"))
+    buckets = Dict{DateTime, Vector{JSON3.Object}}()
+    for talk in talks
+        start = DateTime(talk.slot.start, DATETIME_FORMAT)
+        push!(get!(() -> [], buckets, start), talk)
+    end
+    return sort(collect(buckets); by=first)
+end
+
+function format_message(talk, color)
+    description = talk.description
+    if length(description) > 2000
+        description = description[1:1994] * " [...]"
+    end
+
+    fields = [(; name="More info", value="https://juliacon2020.now.sh/live")]
```

Comment:
@travigd Should this URL actually be https://live.juliacon.org/live ? I just realized that we were using that URL.

Reply:
Yep, it should. It'll autoredirect but yeah.